js/iframe-coordinator/libs/client.ts

```diff
@@ -1,31 +1,74 @@
 import { Elm } from "../elm/Client.elm";
 import { PublicationHandler } from "./types";
 
-let worker: ClientProgram = null;
-let messageHandlers: Array<PublicationHandler> = [];
+interface ToastOptions {
+  title?: string;
+  custom?: { [x: string]: any };
+}
+
+class Client {
+  private _worker: ClientProgram;
+  private _messageHandlers: Array<PublicationHandler> = [];
+
+  start(): void {
+    if (!this._worker) {
+      this._worker = Elm.Client.init();
+  
+      window.addEventListener("message", (event: MessageEvent) => {
+        this._worker.ports.fromHost.send({
+          origin: event.origin,
+          data: event.data
+        });
+      });
+  
+      this._worker.ports.toHost.subscribe((message: any) => {
+        window.parent.postMessage(message, "*");
+      });
+  
+      this._worker.ports.toClient.subscribe((message: any) => {
+        if (message.msgType == "publish") {
+          this._messageHandlers.forEach(handler => {
+            handler(message.msg);
+          });
+        }
+      });
```

Comment:
Do you think it would improve the API to handle the call to `Elm.Client.init()` in the constructor, along with the port setup, and then maybe call this something like `startLinkHandler`?  I don't think there's a real benefit to deferring the Elm setup steps, but I could see a case where a client app might not want to activate the link handling by default.

Reply:
I think it makes sense to leave any eventing behind some sort of init/start method.  People may want to defer the iframe from doing anything for a certain trigger.

My ideal API, i think would look something like this:
import { Client } from 'iframe-coordinator';

if (enabled) {
  client.start({ linkHandlers: true });
}

onDisabled() {
  client.stop();
}


either that or Client doesn't return an object, but a subscription func:
import { client } from 'iframe-coordinator';

if (enabled)
  client({ link:handlers: true });

Reply:
Mentioned something similar below, but my vote is:

1. Export the class
1. high-level config can be passed into the constructor
1. init/start function to delay execution as desired (fine if we put listening type config here and not in the constructor)

Reply:
I'm fine with keeping all of the initialization in the `start` method if ya'll think it's better that way. My initial comment was made assuming that `client.ts` exported the Client constructor, not an instance. I missed that detail the first time around. I do think exporting the class rather than just an instance is a good idea. That lets consumers extend it if they want. If we want to enforce a singleton client instance, we could do that within the constructor.

Reply:
sounds good to me.  I didn't know if there was a reason we made it a singleton in the first place.